### ISSUE TITLE
Rewrite application logging

### ISSUE THREAD

**Issue by ColdHeat** [CTFd/CTFd#844]
These logs are a holdover from before CTFd was even public. I have significant doubts they're even used and in many situations they're stuck inside of a transient docker container. @tecknicaltom if you do make use of these logs it would be useful to understand how. 

First thoughts:
 - Get these logs off of the fs by default and into something more inline with how docker does logs 
 - Take better advantage of log levels throughout the application

**Comment by tecknicaltom** [comment]
I was developing a plugin related to login and was looking for the log messages to help to debug it. I didn't use the log files, but since the `log()` function prints as well as writes to file, the messages make their way to the container's log. I wasn't tied to the log files on disk, just wanted to make sure that the points in the code where I saw `log()` called were reached. Looking back, I'm not even sure why I discovered that `init_logs()` wasn't called, because I assume now that the message would have made it into the docker log regardless of if the handler was installed or not cause the print is done right in `log()`.

tl;dr: I've got no use case that would prevent you from removing those log files and refactor the logging code

**Comment by ColdHeat** [comment]
We likely want to rewrite logging to be in JSON versus in the currently unstructured text data. See https://github.com/CTFd/CTFd/pull/2580
